File: src - java/Esercizi/ProfonditaMaggioreDiDue/ProfonditaMaggioreDiDue.java
	public boolean profMaggioreDiDue() {
		return
			(root != null) && // c'è la radice e...
			(
					// ...o esiste il nodo di sinistra e...
					((root.sin != null) && 
							// ...questo ha almeno un figlio...
							(
									(root.sin.sin != null) || 
									(root.sin.des != null)
							))
					
					|| //oppure
					
					// ...esiste il nodo di destra e...
					((root.des != null) &&
							// ...questo ha almeno un figlio.
							(
									(root.des.sin != null) || 
									(root.des.des != null)
							))
			);
	}

File: src - c++/Esercizi/ProfonditaMaggioreDiDue/ProfonditaMaggioreDiDue.cpp
bool AlberoBinario::ProfMaggioreDiDue() const {
  return
    //c'è la radice e...
    //...o esiste il nodo di sinistra e questo ha almeno un figlio
    //oppure esiste il nodo di destra e questo ha almeno un figlio.
    //Tradotto in codice si ha:
    root && (
      (root->sin && (root->sin->sin || root->sin->des)) ||
      (root->des && (root->des->sin || root->des->des))
    );
}
